File: Data Structures & Algorithms/palindrome-number/submission-0.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        long long num=x;
        long long revNum=0;

        while(x!=0){
            int digit=x%10;
            x/=10;
            revNum=(revNum*10)+digit;
        }
        return num==revNum;
    }
};

File: Data Structures & Algorithms/intersection-of-two-arrays/submission-0.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        mp={}
        res_map=defaultdict(int)
        for num in nums1:
            if num not in mp:
                mp[num] = 1
            else:
                mp[num] += 1

        for num in nums2:
            if num in mp:
                res_map[num]+=1
        
        res=[]
        for k,v in res_map.items():
            res.append(k)
        
        return res
